src/clients/client.ts

```diff
@@ -86,20 +86,22 @@ class Client {
    * Get all the indexes as Index instances.
    * @memberof MeiliSearch
    * @method getIndexes
-   * @returns {Promise<Index[]>} Promise returning array of raw index information
+   *
+   * @returns {Promise<Result<Index[]>>} Promise returning array of raw index information
    */
-  async getIndexes(): Promise<Index[]> {
-    const { results } = await this.getRawIndexes()
-    const indexes: Index[] = results.map(
+  async getIndexes(): Promise<Result<Index[]>> {
+    const rawIndexes = await this.getRawIndexes()
+    const indexes: Index[] = rawIndexes.results.map(
       (index) => new Index(this.config, index.uid, index.primaryKey)
     )
-    return indexes
+    return { ...rawIndexes, results: indexes }
```

Comment:
I think the name `rawIndexes` confusing because in my first read I understood that your method would result in this response:


```js
{
   "results": [],
   results: [....]
}
```

I know that's impossible in JS, but in ruby for example isn't (because `results:` is a symbol, and `"results"` is a string a different type of object). Anyway, I think you could call the `rawIndexes` as just response?

Reply:
I'm not sure I understand :/ 

The following will override the `results` key of `rawIndexes` with the new `results` key so that results is an array of `Index()` and not an array of `{ uid: 1 }` 
```
{ ...rawIndexes, results: indexes }
```

RawIndexes returns like this:
```js
{
   results: [ { uid: 1 .. } ], 
   limit: 0, 
   // ...
}
```

and `getIndexes` returns like this
```
{
   results: [ Index(uid: 1), Index(uid: 2)], 
   limit: 0, 
   // ...
}
```